<launch>
  <!-- Interface arguments -->
  <arg name="ns"                   default="tesse"/>
  <arg name="teleop"               default="false"/>
  <arg name="drive_with_ackermann" default="false"/>
  <arg name="params_dir"           default="$(find tesse_ros_bridge)/params"/>
  <arg name="use_gt_frames"        default="true"/>
  <arg name="enable_step_mode"     default="true"/>

  <!-- Network arguments -->
  <arg name="sim_ip"                default="127.0.0.1"/>
  <arg name="self_ip"               default="127.0.0.1"/>
  <arg name="use_broadcast"         default="true"/>

  <!-- Sim arguments -->
  <arg name="use_sim_time"          default="true"/>
  <arg name="frame_rate"            default="20.0"/>
  <arg name="imu_rate"              default="200"/>
  <arg name="scan_rate"             default="40.0"/>
  <arg name="enable_collision"      default="true"/>
  <arg name="initial_scene"         default="1"/>

  <arg name="publish_clock"         default="true"/>

  <arg name="body_frame_id"         default="base_link"/>
  <!-- <arg name="body_frame_id"         default="base_link_kimera"/> -->

  <!-- Trajectory arguments -->
  <arg name="traj_filepath" default="$(find tesse_ros_bridge)/trajectories/traj_office_long.csv"/>
  <arg name="drive_topic"   default="drive"/>
  <arg name="clock_rate"    default="30"/>

  <include file="$(find tesse_ros_bridge)/launch/tesse_bridge.launch" pass_all_args="true">
    <remap from="drive" to="$(arg drive_topic)"/>
  </include>

  <node ns="$(arg ns)" pkg="tesse_ros_bridge" type="traj_loader.py" name="traj_loader" output="screen">
    <param name="traj_filepath"  value="$(arg traj_filepath)"/>
    <param name="drive_topic"    value="$(arg drive_topic)"/>
    <param name="clock_rate"     value="$(arg clock_rate)"/>
  </node>
</launch>
